.github/workflows/haskell.yml

```diff
@@ -96,16 +96,18 @@ jobs:
                           -a ("extra-lib-dirs: {0}, C:/msys64/mingw64/lib" -f $ghcMingwDir) `
                           -f init
 
-    - name: "OUTPUT Record cabal-store (Linux)"
-      id: lin-setup-haskell
-      if: runner.os != 'Windows'
-      run: echo "cabal-store=/home/runner/.cabal/store" >> $GITHUB_OUTPUT
-
-    - name: "OUTPUT Record cabal-store (Windows)"
-      id: win-setup-haskell
-      if: runner.os == 'Windows'
+    - name: "Setup cabal-store"
+      id: cabal-store
       shell: bash
-      run: echo "cabal-store=C:\\cabal\\store" >> $GITHUB_OUTPUT   
+      run: |
+        cabal help user-config
+        cabal_config_file="$(cabal help user-config | grep -A 1 'You can edit the cabal configuration file to set defaults' | tail -n 1 | xargs)"
+
+        if [[ '${{ runner.os }}' != 'Windows' ]]; then
+          echo "cabal-store=$(dirname "$cabal_config_file")/store" | tee -a "$GITHUB_OUTPUT"
+        else
+          echo "cabal-store=C:\\cabal\\store" | tee -a "$GITHUB_OUTPUT"
+        fi
```

Comment:
Unify the computation of the `cabal` store directory to a single step.  This makes referencing the cabal store in later steps easier.

This also fixes the problem where on `MacOS`, the `cabal-store=/home/runner/.cabal/store` is incorrect.  This should be `/Users/runner/.cabal/store`.

To avoid this problem in future, on Linux and MacOS, the step queries `cabal` for the some information to compute the location of the `cabal` store.

Reply:
Can you put this as a comment in the file?